### PR TITLE
T2294

### PR DESCRIPTION
**Pull request by mjschranz**

### CODE REVIEW

File: cornfield/test/lib.user.test.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+test( "updateProject invalid parameters", function( t ) {
+  t.plan( 2 );
+
+  var mockCallback = function( err, project ) {
+        t.ok( true, "callback was successfully hit" );
+        t.ok( err, "Successfully received an error with invalid parameters" );
+
+        t.end();
+        user.closeDBConnection();
```
Comment (jbuck) [inline cornfield/test/lib.user.test.js]: Move this into it's own test block

File: cornfield/test/lib.user.test.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
+var test = require( "tap" ).test,
+    user = require( "../lib/user" ),
+    mockEmail = "test.example.org",
```
Comment (jbuck) [inline cornfield/test/lib.user.test.js]: test@example.org

File: cornfield/test/lib.user.test.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    id;
+
+test( "createProject valid parameters", function( t ) {
+  t.plan( 7 );
+
+  var mockCallback = function( err, project ) {
+        // Store ID for later tests
+        id = project._id;
+
+        t.ok( true, "callback was successfully hit" );
```
Comment (jbuck) [inline cornfield/test/lib.user.test.js]: No need for this, since you have other tests here, and you use t.plan() to set the expected number of tests. Nuke it from the other tests as well.

File: cornfield/test/lib.user.test.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        t.equal( project.author, mockData.author, "Properly Set Author of Project" );
+        t.equal( project.template, mockData.template, "Properly Set Template of Project" );
+
+        t.end();
+      };
+
+  user.createProject( mockEmail, mockData, mockCallback );
+});
+
+test( "createProject invalid parameters", function( t ) {
```
Comment (jbuck) [inline cornfield/test/lib.user.test.js]: Add a test for a null email as well

File: cornfield/test/lib.user.test.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+  t.plan( 2 );
+
+  var mockCallback = function( err, project ) {
+        t.ok( true, "callback was successfully hit" );
+        t.ok( err, "Successfully received an error with invalid parameters" );
```
Comment (jbuck) [inline cornfield/test/lib.user.test.js]: Check the specific error message

File: cornfield/test/lib.user.test.js [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+  user.createProject( mockEmail, mockData, mockCallback );
+});
+
+test( "findAllProjects valid parameters", function( t ) {
+  t.plan( 2 );
+
+  var mockCallback = function( err, docs ) {
+        t.ok( true, "callback was successfully hit" );
+        t.ok( docs, "Successfully returned all projects for " + mockEmail );
```
Comment (jbuck) [inline cornfield/test/lib.user.test.js]: maybe check number of returned documents, whether it's an array?
